src/core/query.js

```diff
@@ -778,9 +778,9 @@ export class Query {
             const bField = nested(b, field);
 
             if (isDefined(aField) && isDefined(bField) === false) {
```

Comment:
Wouldn't this treat `null` and `undefined` the same? I'm not sure what the spec is - what should happen to objects, which do not posses the field? Regardless of that, wouldn't this result in an unpredictable order among entities for which `isDefined()` returns false?

Reply:
Yes this does treat `null` and `undefined` the same (or appears too). When I tested it with `null` and `undefined`, `null` came first in an ascending order and `undefined` came first in a descending order.

The same order would be returned depending on if the sort is `descending` or `ascending` and `isDefined()` returned `false`.

Reply:
Isn't that because of the order in the initial array? Because comparing `null` and `undefined` always returns `false`. And what is the desired behaviour here? We are trying to align it to the server's, right - what is that?

Reply:
I will have to see what the server returns for these different use cases. I don't think there is a spec somewhere to reference.

Reply:
For a sort on `_id` in descending order the server returns

```json
[ 
  { _id: 2, customProperty: 'Xlv3a36WdCahHscBdu' },
  { _id: 1, customProperty: 'UbiUgNfxLdiHrX8vOS' },
  { _id: null, customProperty: 'ul29WKATNXhGD9SUwF' }, // _id is null
  { customProperty: 'yRNvuXJ79vC3OTW9HZ' } // _id is undefined
]
```

For a sort on `_id` in ascending order the server returns

```json
[ 
  { customProperty: 'uhljZH0VRHUNbW9Ck2' }, // _id is undefined
  { _id: null, customProperty: 'YCtYEPDQYyPtnN8n3E' }, // _id is null
  { _id: 1, customProperty: 'oYYRouQ9eZ026J3pKq' },
  { _id: 2, customProperty: 'gUpApX6uCIjYg8FTx8' } 
]
```

Reply:
How did you create an entity without an _id? I tested it with a different field and null always came up before objects which do not posses the field in ascending order - in both mongodb and for requests to Kinvey.

Reply:
Ah, nevermind, turns out it's just the natural order that's at play here.

Reply:
The natural order is preserved with the latest changes.